modules/ROOT/pages/dataweave-maven-plugin.adoc

```diff
@@ -0,0 +1,329 @@
+= Packaging and Deploying DataWeave Libraries
+
+Use the DataWeave Maven plugin to integrate the packaging and deployment of your DataWeave libraries with your Maven lifecycle.
+
+The DataWeave Maven plugin has six main goals you can run:
+
+* `compile` +
+Compiles the source code for your DataWeave library.
+
+* `test` +
+Tests the compiled source code using the https://github.com/mulesoft/data-weave-testing-framework[DataWeave Testing Framework].
+
+* `data-weave:generate-docs` +
+Automatically generates the documentation for your DataWeave library.
+
+* `package` +
+Packages the compiled code in a `JAR` file.
+
+* `data-weave:deploy-docs` +
+Uploads the auto-generated documentation for your DataWeave library to Anypoint Exchange.
+
+* `deploy` +
+Automatically uploads your DataWeave library to the deployment target and uploads the auto-generated documentation to Exchange.
+
+== Add the DataWeave Maven Plugin to a Project
+
+To enable the DataWeave Maven plugin in your project:
+
+* Add the following Maven dependency to your project's `pom.xml` file:
++
+[source,xml,linenums]
+----
+<plugin>
+include::dataweave::partial$dataweave-maven-plugin-config.adoc[tag=config]
+</plugin>
+----
+
+* Add the following repositories to your project's `pom.xml` file::
++
+[source,xml,linenums]
+----
+<pluginRepositories>
+    <pluginRepository>
+        <id>mule-releases</id>
+        <url>https://repository-master.mulesoft.org/nexus/content/repositories/releases</url>
+    </pluginRepository>
+    <pluginRepository>
+        <id>mule-snapshots</id>
+        <url>https://repository-master.mulesoft.org/nexus/content/repositories/snapshots</url>
+    </pluginRepository>
+</pluginRepositories>
+----
+
+== Compile a DataWeave Library
+
+Run the `compile` goal to compile your DataWeave library from source code, for example:
```

Comment:
Why are these called "goal"s?  
A goal is typically the end toward which effort is directed, not the effort itself; more specifically, a goal is an idea of the future or desired result that a person or a group of people envision, plan, and commit to achieve.
I would globally replace this term with something more specific to commands; will stop marking.